build/Microsoft.DotNet.Cli.GitCommitInfo.targets

```diff
@@ -0,0 +1,44 @@
+<Project ToolsVersion="15.0">
+  <Target Name="WriteGitCommitInfoProps"
+          Condition=" !Exists('$(GitCommitInfoProps)') "
+          DependsOnTargets="BuildDotnetCliBuildFramework">
+    <GetCommitHash RepoRoot="$(RepoRoot)">
+      <Output TaskParameter="CommitHash" PropertyName="GitInfoCommitHash" />
+    </GetCommitHash>
+
+    <GenerateBuildVersionInfo RepoRoot="$(RepoRoot)"
+                              VersionMajor="$(VersionMajor)"
+                              VersionMinor="$(VersionMinor)"
+                              VersionPatch="$(VersionPatch)"
+                              ReleaseSuffix="$(ReleaseSuffix)">
+      <Output TaskParameter="CommitCount" PropertyName="GitInfoCommitCount" />
+      <Output TaskParameter="MsiVersion" PropertyName="BuildInfoMsiVersion" />
+      <Output TaskParameter="VersionBadgeMoniker" PropertyName="BuildInfoVersionBadgeMoniker" />
+    </GenerateBuildVersionInfo>
+
+    <PropertyGroup>
+      <ShouldOverWriteThePropsFile 
+        Condition=" '$(CommitHash)' != '$(GitInfoCommitHash)' Or 
+                    '$(CommitCount)' != '$(GitInfoCommitCount)' Or 
+                    '$(MsiVersion)' != '$(BuildInfoMsiVersion)' Or 
+                    '$(VersionBadgeMoniker)' != '$(BuildInfoVersionBadgeMoniker)' ">true</ShouldOverWriteThePropsFile>
+
+      <GitCommitInfoPropsContent>
+&lt;Project ToolsVersion=&quot;15.0&quot;&gt;
+    &lt;PropertyGroup&gt;
+        &lt;CommitHash&gt;$(GitInfoCommitHash)&lt;/CommitHash&gt;
+        &lt;CommitCount&gt;$(GitInfoCommitCount)&lt;/CommitCount&gt;
+        &lt;MsiVersion&gt;$(BuildInfoMsiVersion)&lt;/MsiVersion&gt;
+        &lt;VersionBadgeMoniker&gt;$(BuildInfoVersionBadgeMoniker)&lt;/VersionBadgeMoniker&gt;
+    &lt;/PropertyGroup&gt;
+&lt;/Project&gt;
+      </GitCommitInfoPropsContent>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(GeneratedPropsDir)" />
```

Comment:
This needs to be conditional, and it should be extracted to a target for sharing amongst these two targets.